File: data-server-provide.cpp
#include <stdlib.h>
#include <string>

using namespace std;

int main(int argc, char** argv) {
	if(argc == 2) {
		string cmd = string("env PORT=") + string(argv[1]) + string(" node data-server-provide.js");
		system(cmd.c_str());
		return 1;
	}
	else return -1;
}

File: data-server-provide.js
const http = require("http");
const { spawn } = require("child_process");

const server = http.createServer((req, res) => {
	res.setHeader('Content-Type', 'text/plain');
	if(req.url == "/genRandom") {
		const rdGr = spawn("./genRand");
		rdGr.stdout.on('data', d => {
			res.write(d);
		});
		rdGr.on('exit', () => {
			res.end();
			console.log("close");
			server.close();
		});
	}
	else {
		res.end("Can't recongnize");
	}
});

server.listen(parseInt(process.env.PORT));

